File: html-to-template-converter/models/globals.js
const mongoose = require("mongoose");

const GlobalTemplateSchema = new mongoose.Schema(
  {
    block_id: { type: mongoose.Schema.Types.ObjectId, required: true },
    name: { type: String, required: true },
    layout_id: { type: Object, required: true },
    settings: {
      type: Object,
      default: {
        paddings: {
          top: 80,
          bottom: 80,
        },
        background: {
          type: "color",
          background_color_id: null,
        },
      },
    },
    data: { type: Object, required: true, default: {} },
    createdAt: { type: Date, default: Date.now },
    access: {
      type: Object,
      default: { duplicate: false, delete: true, sortable: true },
    },
    unique: { type: Boolean, default: false },
    location: {
      type: String,
      enum: ["header", "footer", "page"],
      default: "page",
    },
  },
  { collection: "globals", versionKey: false }
);

module.exports = mongoose.model("GlobalModel", GlobalTemplateSchema);

File: backend/Models/Globals.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace DtcBackend.Models;

public class Globals
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string id { get; set; } = null!;
    [BsonRepresentation(BsonType.ObjectId)]
    public string block_id { get; set; } = null!;

    [BsonElement("name")]
    public string name { get; set; } = null!;

    [BsonRepresentation(BsonType.ObjectId)]
    public string layout_id { get; set; } = null!;

    [BsonElement("settings")]
    public BlockSettings settings { get; set; } = new();

    [BsonElement("data")]
    public Dictionary<string, object> data { get; set; } = new();

    [BsonElement("access")]
    public BlockAccess access { get; set; } = new();

    [BsonElement("unique")]
    public bool unique { get; set; } = false;

    [BsonElement("location")]
    public string location { get; set; } = "page"; // "header", "footer", "page"

    [BsonElement("createdAt")]
    public DateTime createdAt { get; set; } = DateTime.UtcNow;

    [BsonElement("updatedAt")]
    public DateTime updatedAt { get; set; } = DateTime.UtcNow;
}